lib/ansible/modules/cloud/vmware/vmware_vm_host_drs_rule.py

```diff
@@ -0,0 +1,467 @@
+            vm_group_name: Name of virtual machine group
+
+        Returns: True if Group is created, False if not.
+
+        """
+        vm_group_created = False
+        if cluster_obj is None:
+            cluster_obj = self.cluster_obj
+
+        vm_group_spec = vim.cluster.VmGroup()
+        vm_group_spec.name = vm_group_name
+        vm_group_spec.vm = vms_obj_list
+
+        cluster_group_spec = vim.cluster.GroupSpec()
+        cluster_group_spec.operation = 'add'
+        config_spec = vim.cluster.ConfigSpecEx()
+        config_spec.groupSpec.append(cluster_group_spec)
+
+        try:
+            vm_group_task = cluster_obj.ReconfigureEx(spec=config_spec, modify=True)
+            vm_group_created, result = wait_for_task(vm_group_task)
+        except vmodl.fault.InvalidRequest as e:
+            result = to_native(e.msg)
+        except Exception as e:
+            result = to_native(e)
+
+        return vm_group_created, result
+
+    def get_rule_key_by_name(self, cluster_obj=None, rule_name=None):
+        """
+        Function to get a specific DRS rule key by name
+        Args:
+            rule_name: Name of rule
+            cluster_obj: Cluster managed object
+
+        Returns: Rule Object if found or None
+
+        """
+        if cluster_obj is None:
+            cluster_obj = self.cluster_obj
+
+        if rule_name:
+            rules_list = [rule for rule in cluster_obj.configuration.rule if rule.name == rule_name and isinstance(rule, vim.cluster.VmHostRuleInfo)]
+            if rules_list:
+                return rules_list[0]
+        # No rule found
+        return None
+
+    @staticmethod
+    def normalize_rule_spec(rule_obj=None):
+        """
+        Function to return human readable rule spec
+        Args:
+            rule_obj: Rule managed object
+
+        Returns: Dictionary with Rule info
+
+        """
+        if rule_obj is None:
+            return {}
+        return dict(rule_key=rule_obj.key,
+                    rule_enabled=rule_obj.enabled,
+                    rule_name=rule_obj.name,
+                    rule_mandatory=rule_obj.mandatory,
+                    rule_uuid=rule_obj.ruleUuid,
+                    rule_vm_group_name=rule_obj.vmGroupName,
+                    rule_affine_host_group_name=rule_obj.affineHostGroupName,
+                    rule_anti_affine_host_group_name=rule_obj.antiAffineHostGroupName,
+                    )
+
+    # Create
+    def create(self):
+        """
+        Function to create a DRS rule if rule does not exists
+        """
+        rule_obj = self.get_rule_key_by_name(rule_name=self.rule_name)
+        if rule_obj is not None:
+            # Rule already exists, remove and create again
+            # Cluster does not allow editing existing rule
+            existing_rule = self.normalize_rule_spec(rule_obj=rule_obj)
+            if ((existing_rule['rule_vm_group_name'] == self.vm_group_name) or
```

Comment:
Should add a check for rule_(affine/anti_affine)_host_group_name?

Reply:
Done.

Reply:
One more question on this: The checks are to see if the current VM group name _equals_ the new VM group name? (same for hosts), shouldn't that be if they differ?
It would make the code:
```
if ((existing_rule['rule_vm_group_name'] != self.vm_group_name) or
    (existing_rule['rule_anti_affine_host_group_name'] != self.host_group_name and not self.affinity_rule) or
    (existing_rule['rule_affine_host_group_name'] != self.host_group_name and self.affinity_rule) or
    (existing_rule['rule_enabled'] != self.enabled) or
    (existing_rule['rule_mandatory'] != self.mandatory))
```
(added check to only check for the right host group name depending on type of rule)

Reply:
idem, but I would have write:
```
        rule_obj = self.get_rule_key_by_name(rule_name=self.rule_name)
        if rule_obj is not None:
            # Rule already exists, remove and create again
            # Cluster does not allow editing existing rule
            existing_rule = self.normalize_rule_spec(rule_obj=rule_obj)
            if ((existing_rule['rule_vm_group_name'] == self.vm_group_name) or
                    (existing_rule['rule_anti_affine_host_group_name'] == self.host_group_name) or
                    (existing_rule['rule_affine_host_group_name'] == self.host_group_name) or
                    (existing_rule['rule_enabled'] == self.enabled) or
                    (existing_rule['rule_mandatory'] == self.mandatory)):
                # Rule is same as existing rule, evacuate
                self.module.exit_json(changed=False, result=existing_rule)
            # Delete existing rule as we cannot edit it
            changed, result = self.delete(rule_name=self.rule_name)
            if not changed:
                self.module.fail_json(msg="Failed to delete while updating rule %s due to %s" % (self.rule_name, result))
        changed, result = self.create_rule_spec()
        return changed, result
```